File: app/src/main/java/com/cdiamon/autocolorist/fragments/GalleryFragment.java
package com.cdiamon.autocolorist.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.cdiamon.autocolorist.R;
import com.cdiamon.autocolorist.explistview.ActivitySearchGallery;

public class GalleryFragment extends Fragment {

    public static GalleryFragment newInstance() {
        return new GalleryFragment();
    }

    public GalleryFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
//        return inflater.inflate(R.layout.fragment_gallery, container, false);
        View view = inflater.inflate(R.layout.fragment_gallery, container, false);

        Button goButton = (Button) view.findViewById(R.id.fragm_button_go);
        goButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent;
                intent = new Intent(GalleryFragment.this.getContext(), ActivitySearchGallery.class);
                Toast.makeText(GalleryFragment.this.getContext(), R.string.fragmentGalleryToastGo, Toast.LENGTH_SHORT).show();
                startActivity(intent);
            }
        });

        return view;
    }
}

File: app/src/main/java/com/cdiamon/autocolorist/fragments/GalleryFragment.kt
package com.cdiamon.autocolorist.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.cdiamon.autocolorist.R
import com.cdiamon.autocolorist.databinding.FragmentGalleryBinding
import com.cdiamon.autocolorist.explistview.SearchGalleryActivity

class GalleryFragment : Fragment() {

    private var _binding: FragmentGalleryBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        _binding = FragmentGalleryBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.goButton.setOnClickListener {
            val intent = Intent(context, SearchGalleryActivity::class.java)
            Toast.makeText(context, R.string.fragmentGalleryToastGo, Toast.LENGTH_SHORT).show()
            startActivity(intent)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        fun newInstance(): GalleryFragment {
            return GalleryFragment()
        }
    }
}
